microbiomedata/nmdc-server
# Ingest COG terms, pathways and functions

Fix #1333

### Changes
In additional to populating postgres tables with IDs, definitions of, and relationships between KEGG modules, pathways, and orthologs, ingest now populates those tables with COG terms, functions and pathways. The files used to ingest this information are found in the NERSC CFS. From the NMDC home directory (`/global/cfs/cdirs/m3408/`) they can be found under `/ingest/cog/`.

New models and an associated migration have also been included. The `ko_term_to_text` table has been overloaded to include a mapping of ID to description for COG terms/pathways/functions as well. Note that COG pathways don't have a unique identifier associated with them, so the description is used for both the `term` and `text` column.

Additionally, documents in the mongo collection `functional_annotation_agg` whose `gene_function_id` begins with `COG` will be ingested.

### To test
To test locally you must modify the setup you normally use for local ingest to make sure the necessary files are mounted. See `ingest/kegg.py` for some additional details. Additionally, you must make sure that there are some representative documents in your `functional_annotation_agg` collection. After running local ingest, inspect the relevant tables to make sure that COG terms, functions and pathways are ingested, along with the `functional_annotation_agg` documents.

### Deployment Notes
I've rearranged the files in NERSC so that all of the files that ingest needs are in `.../m3408/ingest/`. I think we should update our mount points to mount to a subdirectory of `.../m3408` instead of mounting to the whole entire project directory.
